jmbitcoin/jmbitcoin/secp256k1_transaction.py

```diff
@@ -508,28 +575,30 @@ def verify_tx_input(tx, i, script, sig, pub, witness=None, amount=None):
     if re.match('^[0-9a-fA-F]*$', script):
         script = binascii.unhexlify(script)
     if not re.match('^[0-9a-fA-F]*$', sig):
-        sig = safe_hexlify(sig)
+        sig = binascii.hexlify(sig)
     if not re.match('^[0-9a-fA-F]*$', pub):
-        pub = safe_hexlify(pub)
+        pub = binascii.hexlify(pub)
     if witness:
         if not re.match('^[0-9a-fA-F]*$', witness):
             witness = safe_hexlify(witness)
-    hashcode = decode(sig[-2:], 16)
+    hashcode = binascii.unhexlify(sig[-2:])
     if witness and amount:
         #TODO assumes p2sh wrapped segwit input; OK for JM wallets
-        scriptCode = "76a914"+hash160(binascii.unhexlify(pub))+"88ac"
-        modtx = segwit_signature_form(deserialize(binascii.hexlify(tx)), int(i),
-                                      scriptCode, amount, hashcode)
+        scriptCode = binascii.unhexlify("76a914"+hash160(binascii.unhexlify(pub))+"88ac")
```

Comment:
if it's unhexed anyway it could use bytes + bin_hash160

Reply:
this might actually be a good opportunity to move the code in cryptoengine to jmbitcoin: https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/7e22cbe310f3f04ac59faeebca3924b1540e1735/jmclient/jmclient/cryptoengine.py#L24-L51

Reply:
I didn't really want to refactor this sort of things too much yet since I want to eventually replace transaction parsing with `python-bitcoinlib`.